packages/core/test/proxy-scripts/lsp-broker/LspUniswapV2Broker.js

```diff
@@ -496,4 +467,395 @@ contract("LspUniswapV2Broker", function (accounts) {
+      const longPerLp = (await longToken.balanceOf(pair.address)).mul(adjustment).div(await lpToken.totalSupply());
+      const shortPerLp = (await shortToken.balanceOf(pair.address)).mul(adjustment).div(await lpToken.totalSupply());
+
+      // Next, the trader removes their liquidity to validate they get back the right number of long/short tokens.
+      await lpToken.approve(router.address, MAX_UINT_VAL, { from: trader });
+
+      const traderLpBalance = await lpToken.balanceOf(trader);
+
+      await router.removeLiquidity(
+        longToken.address,
+        shortToken.address,
+        traderLpBalance.toString(),
+        "0",
+        "0",
+        trader,
+        MAX_UINT_VAL,
+        { from: trader }
+      );
+
+      // Trader should get back tokens in the exact ratio that the pair has between long and short tokens. This should be
+      // approximately equal to a 100:1 ratio, as this is the rate the pool was seeded at (error introduced by single
+      // sided deposit that will make this not exactly 100:1) and should be exactly equal to the ratio from the previous calc.
+      // In these calculations we check that the output is within 0.01 wei of the expected value.
+      assert.equal((await lpToken.balanceOf(trader)).toString(), toWei("0"));
+
+      const longTraderBalance = await longToken.balanceOf(trader);
+      const shortTraderBalance = await shortToken.balanceOf(trader);
+
+      assert.isTrue(
+        longTraderBalance.add(toBN(toWei("0.01"))).gt(longPerLp.mul(traderLpBalance).div(adjustment)) &&
+          longTraderBalance.sub(toBN(toWei("0.01"))).lt(longPerLp.mul(traderLpBalance).div(adjustment))
+      );
+      assert.isTrue(
+        shortTraderBalance.add(toBN(toWei("0.01"))).gt(shortPerLp.mul(traderLpBalance).div(adjustment)) &&
+          shortTraderBalance.sub(toBN(toWei("0.01"))).lt(shortPerLp.mul(traderLpBalance).div(adjustment))
+      );
+
+      // Finally, the ratio of long to short tokens that the trader gets back should be exactly equal to the ratio of
+      // long to short tokens in the pool post addition of tokens. Validate this is the case withing a 10 wei error on
+      // each side. This verifies that the number returned is correctly proportional to that deposited.
+      const walletRatio = longTraderBalance.mul(adjustment).div(shortTraderBalance);
+      const poolRatio = poolLongBalPostAdd.mul(adjustment).div(poolShortBalPostAdd);
+
+      assert.isTrue(walletRatio.addn(10).gte(poolRatio) && walletRatio.subn(10).lte(poolRatio));
+    });
+    it("Can correctly mint and LP in one transaction with pool in unequal ratio with long<short", async function () {
+      // Add 100000 long and 1000000 short. this makes the price 100000/1000000 long/short. In other words every short is
+      // worth 0.01 long. When LPing we will be market selling longs for shorts to reach the appropriate ratio.
+      await addLiquidityToPool(longToken, shortToken, toWei("100000"), toWei("1000000"));
```

Comment:
Can you make the long side initial reserves = 1,000, the amount of longs you mint initially? I'm curious what this "max slippage" situation would look like.

Reply:
Oh nevermind I see you do this test below!